src/contracts/GPv2Settlement.sol

```diff
@@ -290,6 +290,25 @@ contract GPv2Settlement {
         filledAmount[uid] = currentFilledAmount;
     }
 
+    /// @dev Allows settlment function to make arbitrary contract executions.
+    /// @param interaction contains address and calldata of the contract interaction.
+    function executeInteraction(GPv2Encoding.Interaction memory interaction)
+        internal
+        returns (bytes memory response)
+    {
+        // To prevent possible attack on user funds, we explicitly disable
+        // interactions with AllowanceManager contract.
+        require(
+            interaction.target != address(allowanceManager),
+            "GPv2: forbidden interaction"
+        );
+        bool success;
```

Comment:
It's interesting that we are not verifying that the call actually succeeds. I think this is an ok decision (etherscan will still tell us which sub-calls fail). Still, I'd point out that this can lead to us settling batches with e.g. one order and one Uniswap interaction where the interaction failed (possibly accidentally sacrificing part of our collected protocol fees).

Have we considered requiring interactions to be "lax" enough that they have to succeed in order for everything to succeed?

Reply:
I also think this should revert.

Reply:
So shall we include a `require(success)` around here?

Reply:
I have included the following line at the end:

```
require(success, "GPv2: failed interaction");
```